core/rip7560_abi.go

```diff
@@ -96,3 +96,82 @@ func abiDecodeAcceptPaymaster(input []byte, allowSigFail bool) (*AcceptPaymaster
 	}
 	return acceptPaymasterData, err
 }
+
+func abiEncodeRIP7560TransactionEvent(
+	aatx *types.Rip7560AccountAbstractionTx,
+	success bool,
+	actualGasCost *big.Int,
+	actualGasUsed uint64,
+) (topics []common.Hash, data []byte, error error) {
+	id := Rip7560Abi.Events["RIP7560TransactionEvent"].ID
+	paymaster := aatx.Paymaster
+	if paymaster == nil {
+		paymaster = &common.Address{}
+	}
+	deployer := aatx.Deployer
+	if deployer == nil {
+		deployer = &common.Address{}
+	}
+	inputsAll := Rip7560Abi.Events["RIP7560TransactionEvent"].Inputs
+	inputs := inputsAll[3:] // todo: hack - hard-coding the "skipping" of indexed parameters
```

Comment:
hacking 3 times probably means you need PackEvent (and use `input.Indexed` to pack as data or as topic)
I'm surprised there isn't a standard one...